src/inlineautoformatengine.js

```diff
@@ -198,6 +198,7 @@ export default class InlineAutoformatEngine {
 				// Remove delimiters.
 				for ( const range of rangesToRemove ) {
```

Comment:
There are two arrays with ranges: one containing ranges to be formatted and one with ranges to be removed. You're only detaching ranges to remove. I think that we should detach `rangesToFormat` too as they're not longer needed.

Reply:
Yep, looks like one more `LiveRange` is GC'ed:
Before (upcoming change):
![selection_003](https://user-images.githubusercontent.com/247363/31654118-42689468-b325-11e7-952e-7a170714af2c.png)

After:
![selection_004](https://user-images.githubusercontent.com/247363/31654122-469af30a-b325-11e7-9752-d528c1a33668.png)

ps.: I was wondering if this shoudln't be done by Bold (and others) but as I undarstand this Bold leaves bolded range (selection) but here we're not using that range anymore - it's created only to bold text between marks.